crowdnotifier-sdk/src/main/java/org/crowdnotifier/android/sdk/storage/VenueVisitStorage.java

```diff
@@ -49,21 +49,28 @@ public static synchronized VenueVisitStorage getInstance(Context context) {
 		return instance;
 	}
 
-	public long addEntry(EncryptedVenueVisit newVenueVisit) {
+	/**
+	 * Sets a single ID to all entries in the passed list and stores all entries.
+	 * @return the ID
+	 */
+	public long addEntries(List<EncryptedVenueVisit> newVenueVisits){
 		List<EncryptedVenueVisit> venueVisitList = getEntries();
 		long newId = getMaxId(venueVisitList) + 1;
-		newVenueVisit.setId(newId);
-		venueVisitList.add(newVenueVisit);
+		for (EncryptedVenueVisit newVenueVisit : newVenueVisits){
+			newVenueVisit.setId(newId);
+			venueVisitList.add(newVenueVisit);
+		}
 		saveToPrefs(venueVisitList);
 		return newId;
 	}
 
-	public boolean updateEntry(EncryptedVenueVisit newVenueVisit) {
+	public boolean updateEntries(List<EncryptedVenueVisit> newVenueVisits) {
+		if (newVenueVisits == null || newVenueVisits.isEmpty()) return false;
 		List<EncryptedVenueVisit> venueVisitList = getEntries();
-		EncryptedVenueVisit oldEntry = getVenueVisitWithId(venueVisitList, newVenueVisit.getId());
-		if (oldEntry == null) return false;
-		venueVisitList.remove(oldEntry);
-		venueVisitList.add(newVenueVisit);
+		List<EncryptedVenueVisit> oldEntries = getVenueVisitsWithId(venueVisitList, newVenueVisits.get(0).getId());
```

Comment:
Yes in my tests at least it works well. What exactly here concerns you?

Reply:
I was confused here too. Maybe we could think about changing this. The problem is, that one venue visit consists of a List of EncryptedVenueVisisits. We store the List of venue visits as List of EncryptedVenueVisits, so we flatten things out. Maybe it would be easier understandable to store it as a List of Lists, or even a List of a custom class that has a List of EncryptedVenueVisits. But this makes the code at later points a bit more ugly since we have to flatten things out at a later point to check against all EncryptedVenueVisits. But the code here has a lot of implicit assumptions which could lead to problems in the future. (all entries in newVeneuVisits have the same ID etc.)

Reply:
True. I will improve this in a separate PR.